metacat-main/src/main/java/com/netflix/metacat/main/manager/ConnectorManager.java

```diff
@@ -238,4 +328,14 @@ public ConnectorTypeConverter getTypeConverter(final String connectorType) {
     public ConnectorInfoConverter getInfoConverter(final String connectorType) {
         return getPlugin(connectorType).getInfoConverter();
     }
+
+    /**
+     * A Holder class holding the catalog's config and connector factory.
+     */
+    @Data
+    @AllArgsConstructor
+    private static class CatalogHolder {
```

Comment:
I feel we should factor out the catalogRelated operation into a new class, separating from the database/table/part services related informations. The catalog class has 
`private final Table<String, String, CatalogHolder> catalogs = HashBasedTable.create();`
and catalog related information.

Reply:
The catalogHolder is not needed if the table has four columns: catalogName, databaseName, config, connectorFactory